pa10_KA:
  # Publish all joint states -----------------------------------
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 100  

  # Impedance Controller ---------------------------------------
  impedance_controller:
    type: pa10_impedance_control/ImpedanceController
    joints: 
        - S1
        - S2
        - E1
        - E2
        - W1
        - W2
        - W3
    ft_sensor:
        - ft_sensor
    k_gain: 
        - 600.0 #if gazebo's loop rate is the default (600, 210, 80, 50)
        - 210.0
    d_gain: 
        - 800.0
        - 500.0
    chain:
        - world
        - link_wrist_3

  # Admittance Controller ---------------------------------------
  admittance_controller:
    type: pa10_impedance_control/AdmittanceController
    joints: 
        - S1
        - S2
        - E1
        - E2
        - W1
        - W2
        - W3
    ft_sensor:
        - ft_sensor
    k_gain: 
        - 12 #@30Hz #60.0 #if gazebo's loop rate is the default
        - 4.2        #21.0
    d_gain: 
        - 45        #80.0
        - 30.0        #50.0
    chain:
        - world
        - link_wrist_3

  velocity_controller:
    #type: position_controllers/JointGroupPositionController
    type: "velocity_controllers/JointTrajectoryController"
    #type: "position_controllers/JointTrajectoryController"
    #type: "effort_controllers/JointTrajectoryController"

    joints:
        - S1
        - S2
        - E1
        - E2
        - W1
        - W2
        - W3
    gains: 
        S1: {p: 10.0, i: 0.2, d: 0.05, i_clamp: 2.5}
        S2: {p: 10.0, i: 0.2, d: 0.05, i_clamp: 2.5}
        E1: {p: 10.0, i: 0.2, d: 0.05, i_clamp: 2.5}
        E2: {p: 10.0, i: 0.2, d: 0.05, i_clamp: 2.5}
        W1: {p: 10.0, i: 0.2, d: 0.05, i_clamp: 2.5}
        W2: {p: 10.0, i: 0.2, d: 0.05, i_clamp: 2.5}
        W3: {p: 10.0, i: 0.2, d: 0.05, i_clamp: 2.5}

    constraints:
        goal_time: 0.5                    # Override default
        state_publish_rate:  25           # Override default
        action_monitor_rate: 20           # Override default
        stop_trajectory_duration: 0       # Override default









